### ISSUE TITLE
Customizer search module

### ISSUE THREAD

**Issue by abaicus** [Codeinwp/themeisle-companion#124]
Maybe we can build a JS library which we can include in themes to add a search option between customizer options. As we add more options there, the more complicated is to find a setting which you need edit.

**Comment by andreilupu** [comment]
**As a reminder**: There is this plugin https://wordpress.org/plugins/customizer-search
It not even close a thing that I would finally use, but it's a start, and it proves that it can be done.

**Comment by andreilupu** [comment]
I've started to work on this issue, and I have a UI dilemma.
Should we place the search bar top? Or keep it close to the customizer controls?

Here's a mockup compare
https://projects.invisionapp.com/freehand/document/qLP3kdnne

**Comment by ineagu** [comment]
From my perspective this would be very subtle integrated into customizer, like a search icon only, either in the you are customizing or the active theme section. when clikcing on the search icon we will show the search input. The module would be activated by default as well in obfx.

this issue could be proved complex from an UX perspective, I mean how we show the search results, unless we can get a quick good enough solution there is not point to work on it for more than a day.

**Comment by andreilupu** [comment]
I understand. Well, until now I've come to these conclusions:

* I would go with the bottom position since it is a general Customizer control and not a theme specific.
* I would go with a [Select2 autocomplete drop-down](https://select2.org/searching#matching-grouped-options) except that when the user clicks a result, it would instantly focus the setting directly and apply an outline to it.
* It should be a small standalone library which should automatically hook customizer settings. This means that it could eventually be placed in obfx or any plugin.

I find the above solution the simplest way as UI and implementation, but one day is a very ambitious estimation which I would gladly take it as a challenge. 😄 ( <- only if we decide to go this way)

**Comment by ineagu** [comment]
Then for me is a good little thing to work on and see how we can best deliver it to our users.

**Comment by selul** [comment]
@andreilupu yes, the bottom search solution looks amazing for me but i don't know how this will look with the autocomplete list. 
Will be more difficult to filter the customizer sections on the fly, by hiding/showing them, if they match the search query ?

**Comment by andreilupu** [comment]
@selul I find that approach very difficult because the settings are split into panels/sections and it is very hard to display two settings from different panels at the same time. Also, some settings may not be loaded as HTML while searching.
It should be possible, but it would require a lot of extra time.

**Comment by selul** [comment]
@andreilupu then let's keep the autocomplete list.

**Comment by ineagu** [comment]
@selul this can be a good little thing to revive, pixelgrade customify plugin has a solution, astra as well, probably we can go with something simple.

Not sure if another solution won't be to simply recommend this within docs + recommended plugins list: https://wordpress.org/plugins/customizer-search/ ( if is not too intrusive).
